pageserver/src/page_service.rs

```diff
@@ -932,26 +989,58 @@ impl PageServerHandler {
                         }
                     }
                 };
+
+                // Now that we know the shard, set the scope accordingly.
+                let ctx = ctx.with_scope_page_service_pagestream(&shard);
+
```

Comment:
Does this obsolete the messy macro inside the `upgrade_handle_and_set_context` macro?